src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+                                        (s16)this->workf[INNR_CRWN_TX_Y1] & 0x7F, 0x20, 0x20, 1,
+                                        (s16)this->workf[INNR_CRWN_TX_X2] & 0x7F,
+                                        (s16)this->workf[INNR_CRWN_TX_Y2] & 0xFF, 0x20, 0x40));
+        }
+
+        func_80093D18(globalCtx->state.gfxCtx);
+        func_80093D84(globalCtx->state.gfxCtx);
+
+        if (this->work[FOG_TIMER] & 2) {
+            POLY_OPA_DISP = Gfx_SetFog(POLY_OPA_DISP, 255, 50, 0, 0, 900, 1099);
+        } else {
+            POLY_OPA_DISP = Gfx_SetFog(POLY_OPA_DISP, (u32)this->fogR, (u32)this->fogG, (u32)this->fogB, 0,
+                                       this->fogNear, this->fogFar);
+        }
+
+        Matrix_Push();
+        SkelAnime_DrawFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable,
+                              this->skelAnime.dListCount, BossTw_OverrideLimbDraw, BossTw_PostLimbDraw, this);
+        Matrix_Pop();
+        POLY_OPA_DISP = func_800BC8A0(globalCtx, POLY_OPA_DISP);
+    }
+
+    if (this->actor.params == TW_KOTAKE) {
+        if (this->workf[UNK_F9] > 0.0f) {
+            if (this->workf[UNK_F11] > 0.0f) {
+                Vec3f diff;
+                diff.x = this->groundBlastPos2.x - player->actor.world.pos.x;
+                diff.y = this->groundBlastPos2.y - player->actor.world.pos.y;
+                diff.z = this->groundBlastPos2.z - player->actor.world.pos.z;
+
+                if ((fabsf(diff.y) < 10.0f) && ((player->actor.bgCheckFlags & 1) != 0) &&
+                    (sqrtf(SQ(diff.x) + SQ(diff.z)) < (this->workf[UNK_F12] * 4600.0f)) && (sFreezeState == 0) &&
+                    (this->workf[UNK_F11] > 200.0f)) {
+                    sFreezeState = 1;
+                    sTwinrovaPtr->timers[2] = 100;
+                }
+            }
+
+            func_80941BC0(this, globalCtx);
+        }
+    } else {
+        func_80942180(this, globalCtx);
+    }
+
+    if (this->visible) {
+        if (this->actionFunc == BossTw_DeathCS) {
+            func_80943028(&this->actor, globalCtx);
+        } else {
+            func_809426F0(this, globalCtx);
+            Matrix_MultVec3f(&D_8094A9A4, &this->beamOrigin);
+            func_80942C70(&this->actor, globalCtx);
+        }
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_tw.c", 7123);
+}
+
+Gfx* D_8094A9B0[] = {
+    0x0602A9B0,
+    0x0602A070,
+    0x0602A470,
+};
+
+s32 BossTw_TwinrovaOverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                    void* thisx) {
+    BossTw* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_tw.c", 7139);
+
+    switch (limbIndex) {
+        case 21:
+            gSPSegment(POLY_OPA_DISP++, 0xC,
+                       Gfx_TexScroll(globalCtx->state.gfxCtx, 0U, (s16)(f32)this->work[CS_TIMER_1], 8, 8));
```

Comment:
```suggestion
                       Gfx_TexScroll(globalCtx->state.gfxCtx, 0, (s16)(f32)this->work[CS_TIMER_1], 8, 8));
```
there's a bunch of `0U` in this function and in few other places